code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -28,12 +28,20 @@
 /obj/structure/closet/crate/can_close()
 	return TRUE
 
-/obj/structure/closet/crate/open()
+/obj/structure/closet/crate/open(irregular_open = FALSE)
 	if(src.opened)
 		return FALSE
 	if(!src.can_open())
 		return FALSE
 
+	if(!irregular_open)
+		for(var/obj/O in src)
+			if(O.density)
+				var/response = alert(usr, "This crate has been packed with bluespace compression, an item inside won't fit back inside. Are you sure you want to open it?","Bluespace Compression Warning", "No", "Yes")
```

Comment:
Add a check to make sure the person doesnt open the dialogue box then walk away, and remotely open the crate.